tests/optimizer/nonlinear/test_trust_region.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+import theseus as th
+import torch
+
+
+@pytest.mark.parametrize("dof", [1, 8])
+def test_rho(dof):
+    # This test checks that rho = 1 for a simple linear problem min (x - t) ** 2
+    x = th.Vector(tensor=torch.zeros(1, dof), name="x")
+    t = th.Vector(tensor=torch.ones(1, dof), name="t")
+    o = th.Objective()
+    o.add(th.Difference(x, t, th.ScaleCostWeight(1.0), name="cf"))
+    # This is testing TrustRegion base class rather than Dogleg's
+    # Using Dogleg because it's the only subclass of TrustRegion atm
+    opt = th.Dogleg(o)
+    o._resolve_batch_size()
+    opt.linear_solver.linearization.linearize()
+    previous_err = opt._error_metric()
+
+    # Check rho = 1. Predicted error by TrustRegion method should
+    # match actual error after step for a linear problem
+    for _ in range(10):  # repeat a few times with random delta
```

Comment:
Not sure if it is better to first compute `d_gn` direction and then random scale the trust region with `scale * \|d_gn\|`